### PR TITLE
fix(protocol-designer): fix lastModified when exporting

### LINKED ISSUE #6636: bug: (investigate) -- "last exported" info sometimes doesn't show up
# Overview

For some PD, protocols "Last Exported" is blank upon loading them into PD. For others, it shows a date-time.

# Steps to reproduce

~TODO - I have a lot of protocols from never-released dev versions of PD and don't want to confound the results. Try going to an old sandbox (use **release tag** sandbox, not a feature branch or edge plz) and saving protocols then uploading them into `edge`.~

See Max's post in #SPDDRS slack (I don't want to upload here bc it's from a customer)

# Current behavior

Upon loading some protocols:

image

# Expected behavior

Should always show this when you load a protocol

image

### PR DESCRIPTION
**Pull request by IanLondon**
# Overview

Closes #6636

# Changelog

# Review requests

- "Last exported" field should update when the file is saved. If there's a modal or a blocking hint, "last exported" field should update only when the file is actually saved and not just when the Export button is clicked
- "Last exported" should show after importing a file (file must be exported from this branch! Earlier files are missing the field so have no value to show)
- Test code review
- Is middleware approach worth it? The problem with the existing setup was that SAVE_PROTOCOL_FILE is responsible to update lastModified, but there's no way to have the FileSidebar dispatch that, get updated file data info, and then save that fresh info. (Also nice to remove saving implementation from FileSidebar, but middleware isn't the only way to do that). So I figured either 1. use middleware to do this or 2. update lastModified in 2 places or 3. somehow pass a callback that does getState into FileSidebar's save process, so it can dispatch then get fresh state (seems gross). 

Maybe in the future, this saveFileMiddleware can deal with auto-backups too if there's some timer that dispatches "BACKUP_PD_ACTION" every 5 mins if there are changes, or w/e? IDK, point is I feel OK making a new middleware b/c I can brainstorm some additional uses for this middleware outside of this immediate lastModified problem

# Risk assessment

Low, PD-only

### CODE REVIEW

**Review by Kadee80 — APPROVED** [review]
📅 🕐

**Comment by IanLondon** [comment]
Update: this was updating lastModified AFTER save, so it was always wrong and undefined for the first file. Trying a different approach...

**Review by Kadee80 — APPROVED** [review]
🕥 📆

File: protocol-designer/src/components/FileSidebar/index.js [hunk 1]
```diff
@@ -40,18 +40,15 @@ export const FileSidebar: React.AbstractComponent<{||}> = connect<
 )(FileSidebarComponent)
 
 function mapStateToProps(state: BaseState): SP {
-  const protocolName =
-    fileDataSelectors.getFileMetadata(state).protocolName || 'untitled'
+  // const protocolName =
+  //   fileDataSelectors.getFileMetadata(state).protocolName || 'untitled'
```
Comment (shlokamin) [inline protocol-designer/src/components/FileSidebar/index.js]: leftover comments

**Review by shlokamin — COMMENTED** [review]
Can't think of a clean way to do this without middleware, but this seems like a misuse of it doesn't it? We're introducing new middleware to account for this one action, which seems like a code smell.

What we'd like to do is this, right?

1. dispatch action to update the date modified
2. save the file after we know the action has been successfully dispatched and state has been updated (like a callback)

Can redux thunk help us out here at all? 

Could also def be misunderstanding so might be worth chit chatting about this over video call or something

File: protocol-designer/src/load-file/actions.js [hunk 2]
```diff
@@ -90,6 +93,19 @@ export const createNewProtocol = (
...
+
+  const state = getState()
+  const fileData = fileDataSelectors.createFile(state)
+
+  const protocolName =
+    fileDataSelectors.getFileMetadata(state).protocolName || 'untitled'
+  const fileName = `${protocolName}.json`
+
+  saveFile(fileData, fileName)
+}
```
Comment (shlokamin) [inline protocol-designer/src/load-file/actions.js:111]: niiiiice!

**Review by shlokamin — APPROVED** [review]
Good stuff!

**Pull request merged**